src/interfaces.ts

```diff
@@ -0,0 +1,213 @@
+    // null.
+    /* @middlewareOnly */
+    jwtVerifyOptions?: jwt.VerifyOptions,
+    // A comma separated list of strings that give a path in the jwt from which
+    // to extract the postgres role. If none is provided it will use the key
+    // `role` on the root of the jwt.
+    /* @middlewareOnly */
+    jwtRole?: Array<string>,
+    // The Postgres type identifier for the compound type which will be signed as
+    // a JWT token if ever found as the return type of a procedure. Can be of the
+    // form: `my_schema.my_type`. You may use quotes as needed:
+    // `"my-special-schema".my_type`.
+    jwtPgTypeIdentifier?: string,
+    // [DEPRECATED] The audience to use when verifing the JWT token. Deprecated,
+    // use `jwtVerifyOptions.audience` instead.
+    /* @middlewareOnly */
+    jwtAudiences?: Array<string>,
+    // Some one-to-one relations were previously detected as one-to-many - should
+    // we export 'only' the old relation shapes, both new and old but mark the
+    // old ones as 'deprecated', or 'omit' the old relation shapes entirely
+    legacyRelations?: 'only' | 'deprecated' | 'omit',
+    // ONLY use this option if you require the v3 typenames 'Json' and 'Uuid'
+    // over 'JSON' and 'UUID'
+    legacyJsonUuid?: boolean,
+    // Turns off GraphQL query logging. By default PostGraphile will log every
+    // GraphQL query it processes along with some other information. Set this to
+    // `true` to disable that feature.
+    /* @middlewareOnly */
+    disableQueryLog?: boolean,
+    // A plain object specifying custom config values to set in the PostgreSQL
+    // transaction (accessed via `current_setting('my.custom.setting')`) or a
+    // function which will return the same (or a Promise to the same) based on
+    // the incoming web request (e.g. to extract session data)
+    /* @middlewareOnly */
+    pgSettings?: { [key: string]: mixed } | ((req: IncomingMessage) => Promise<{ [key: string]: mixed }>),
+    // Some graphile-build plugins may need additional information available on
+    // the `context` argument to the resolver - you can use this function to
+    // provide such information based on the incoming request - you can even use
+    // this to change the response [experimental], e.g. setting cookies
+    /* @middlewareOnly */
+    additionalGraphQLContextFromRequest?: (req: IncomingMessage, res: ServerResponse) => Promise<{}>,
+    // [experimental] Plugin hook function, enables functionality within
+    // PostGraphile to be expanded with plugins. Generate with
+    // `makePluginHook(plugins)` passing a list of plugin objects.
+    /* @middlewareOnly */
+    pluginHook?: PluginHookFn,
+    // Should we use relay pagination, or simple collections?
+    // "omit" (default) - relay connections only,
+    // "only" - simple collections only (no Relay connections),
+    // "both" - both
+    simpleCollections?: 'omit' | 'both' | 'only',
```

Comment:
We should consider adding `[propName: string]: any` here because PostGraphile plugins can add additional options to this object (e.g. the supporter plugin adds `simpleSubscriptions` as an option)

Reply:
af448cc